src/java.base/share/classes/jdk/internal/classfile/FieldTransform.java

```diff
@@ -0,0 +1,109 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import jdk.internal.classfile.impl.TransformImpl;
+
+/**
+ * A transformation on streams of {@link FieldElement}.
+ *
+ * @see ClassfileTransform
+ */
+@FunctionalInterface
+public non-sealed interface FieldTransform
+        extends ClassfileTransform<FieldTransform, FieldElement, FieldBuilder> {
+
+    /**
+     * A field transform that sends all elements to the builder.
+     */
+    public static final FieldTransform ACCEPT_ALL = new FieldTransform() {
+        @Override
+        public void accept(FieldBuilder builder, FieldElement element) {
+            builder.with(element);
+        }
+    };
+
+    /**
+     * Create a stateful field transform from a {@link Supplier}.  The supplier
+     * will be invoked for each transformation.
+     *
+     * @param supplier a {@link Supplier} that produces a fresh transform object
+     *                 for each traversal
+     * @return the stateful field transform
+     */
+    static FieldTransform ofStateful(Supplier<FieldTransform> supplier) {
+        return new TransformImpl.SupplierFieldTransform(supplier);
+    }
+
+    /**
+     * Create a field transform that passes each element through to the builder,
+     * and calls the specified function when transformation is complete.
+     *
+     * @param finisher the function to call when transformation is complete
+     * @return the field transform
+     */
+    static FieldTransform endHandler(Consumer<FieldBuilder> finisher) {
+        return new FieldTransform() {
+            @Override
+            public void accept(FieldBuilder builder, FieldElement element) {
+                builder.with(element);
+            }
+
+            @Override
+            public void atEnd(FieldBuilder builder) {
+                finisher.accept(builder);
+            }
+        };
+    }
+
+    /**
+     * Create a field transform that passes each element through to the builder,
+     * except for those that the supplied {@link Predicate} is true for.
+     *
+     * @param filter the predicate that determines which elements to drop
+     * @return the field transform
+     */
+    static FieldTransform dropping(Predicate<FieldElement> filter) {
```

Comment:
Could this be a more general static generic method on ClassfileTransform? (but I see CodeTransform doesn't have it - not sure if that's deliberate).

Reply:
It is deliberate as we didn't found meaningful use case for transformation just dropping individual code elements.